### DISCUSSION TITLE
[General] eth_call succeeds but actual transaction fails

### DISCUSSION THREAD

**Discussion by velocore** [zkSync-Community-Hub/zksync-developers#395] [General]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Our v1 -> v2 migration contract has been working fine for several months, but suddenly we have a problem. Nothing has changed on our contract side and eth_call succeeds but still the transaction fails. Could you check if this is a bug and if there's any way we can fix it?

### What is the specific issue or error you're encountering?

I've encountered a strange situation where eth_call succeeds but actual transaction fails.
Our service provides velocore v1 -> v2 migration function. they were working without problem for months. we haven't changed anything recently, then suddenly it stopped working.
this is an example transaction: https://explorer.zksync.io/tx/0x1854990f54db3f76def84ba49c6f5b3cf0e0e4fb24f6beed47d36c441951e678

```javascript
The strange part is that eth_call with the exact same parameter runs without error:
$ cast call --rpc-url zksync 0xf5e67261cb357edb6c7719fefafaab280cb5e2a6 0xd3115a8a000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000020000000000000000000000003cd3e41ff4b95a8ddc915f3c7615ab6f2b478c3301000000000000000000272fbde345771eb0c6adebc54f41a169ff6311fe096f000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000200000000000000000000000003cd3e41ff4b95a8ddc915f3c7615ab6f2b478c33000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000002010000000000000000000000000000000000000000000000000000000000000100010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 --from 0x1234561fEd41DD2D867a038bBdB857f291864225 --gas-limit 100000000

0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000de0b6b3a7640000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff

``` 
all of our migration transactions are failing. everything else is working fine.
the last migration without error happened is this :
https://era.zksync.network/tx/0xa571342dfd04f7edf806883b0e1c9db1a81f7ca641ac9b8291cf87bfe1594ccf
could you please help us? Thank you.

### Can you share the error messages or logs you're receiving, if any?

no error message (Error function_selector = 0x, data = 0x)

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

https://github.com/velocore/velocore-contracts/blob/master/src/pools/vc/VeVC.sol
migration is done through our vault contract and this new veVC contract's execute function. Both haven't been changed since inception.

### Have you tried to isolate the problem, and if so, what were the results?

interacting directly with contract didn't solve the problem.

### What steps have you already taken to try to resolve the issue?

.

**Comment by sfatitetyccindy** [comment]
Issue: Transactions for v1 -> v2 migration suddenly failing, despite successful eth_call.
Possible Causes: Contract bytecode or state may have changed unexpectedly. Gas limit or gas price issues.
Steps Taken: Tried interacting directly with contract, but problem persists.
Next Steps: Review contract bytecode, state, and gas settings for any unexpected changes.

**Reply by velocore** [reply]
Contract bytecode/state hasn't changed at all and not a problem of gas setting.

**Comment by velocore** [comment]
@uF4No Could you help a little bit here?

**Comment by epjocnopasara** [comment]
It seems like there might be an issue with the migration transaction execution, despite successful eth_call. This could be due to various factors such as gas limit, transaction ordering, or contract state. Further debugging and analysis of the contract code, transaction parameters, and environment settings might be necessary to identify and resolve the issue.

**Comment by AlbionaHoti** [comment]
Hi @velocore, we're investigating this one. Apologies for the late answer! 

Can you use the [in-memory node](https://docs.zksync.io/build/test-and-debug/era-test-node.html) to print transaction traces and report back on what you see?
